be/src/exec/pipeline/hashjoin/spillable_hash_join_probe_operator.cpp

```diff
@@ -346,64 +319,103 @@ StatusOr<ChunkPtr> SpillableHashJoinProbeOperator::pull_chunk(RuntimeState* stat
         }
 #endif
     }
+}
 
-    if (_processing_partitions.empty()) {
-        _acquire_next_partitions();
+Status SpillableHashJoinProbeOperator::_restore_probe_partition(RuntimeState* state) {
+    for (size_t i = 0; i < _probers.size(); ++i) {
+        // probe partition has been processed
+        if (_probe_read_eofs[i]) continue;
+        RETURN_IF_ERROR(_current_reader[i]->trigger_restore(
+                state, *_executor,
+                spill::ResourceMemTrackerGuard(tls_mem_tracker, state->query_ctx()->weak_from_this(),
+                                               std::weak_ptr(_current_reader[i]))));
+        if (_current_reader[i]->has_output_data()) {
+            auto chunk_st = _current_reader[i]->restore(
+                    state, *_executor,
+                    spill::ResourceMemTrackerGuard(tls_mem_tracker, state->query_ctx()->weak_from_this(),
+                                                   std::weak_ptr(_current_reader[i])));
+            if (chunk_st.ok() && chunk_st.value() && !chunk_st.value()->is_empty()) {
```

Comment:
for one partition, is _restore_probe_partition only invoked once to fetch a large chunk or is invoked many times to fetch chunk whose number of rows is not greater than chunk size(4096)?

Reply:
the chunk spill to spiller won't exceed chunk_size.
when we spill hash table to spiller. we will make a chunk_slice to do it.
Each time we will get the `chunk_size` row from the hash table
```
    return [this]() -> StatusOr<ChunkPtr> {
        if (_hash_table_build_chunk_slice.empty()) {
            _join_builder->hash_join_builder()->reset(_join_builder->hash_table_param());
            return Status::EndOfFile("eos");
        }
        auto chunk = _hash_table_build_chunk_slice.cutoff(runtime_state()->chunk_size());
        RETURN_IF_ERROR(append_hash_columns(chunk));
        _join_builder->update_build_rows(chunk->num_rows());
        return chunk;
    };
```